src/conv/invokers/impl_gemm.cpp

```diff
@@ -50,27 +50,60 @@ InvokerFactory MakeImplGemmDataInvokerFactory(const ConvolutionContext& ctx)
                     kernel.GetName() == "gridwise_convolution_backward_data_implicit_gemm_v1r1_ncdhw_kczyx_nkdhw"))
                 // clang-format on
                 {
-                    float zero = 0.f;
-                    TensorDescriptor workspaceDesc(
-                        miopenFloat, tensors.outDesc.GetLengths(), tensors.outDesc.GetStrides());
-                    SetTensor(handle, workspaceDesc, workSpace, &zero);
-                    if(handle.IsProfilingEnabled())
-                        elapsed += handle.GetKernelTime();
+                    const auto y           = tensors.wDesc.GetLengths()[2];
+                    const auto x           = tensors.wDesc.GetLengths()[3];
+                    const auto stride_h    = conv.GetConvStrides()[0];
+                    const auto stride_w    = conv.GetConvStrides()[1];
+                    const auto dialation_h = conv.GetConvDilations()[0];
+                    const auto dialation_w = conv.GetConvDilations()[1];
+
+                    if((stride_h >= dialation_h * (y - 1) + 1) &&
+                       (stride_w >= dialation_w * (x - 1) + 1))
+                    {
+                        // If padding exits at any side, needs to initialize with 0s.
+                        if(!miopen::all_of(conv.GetConvPads(), [](auto v) { return v == 0; }))
```

Comment:
You don't need to initialize zero if input has padding. Because padding is implicit and doesn't occupy any memory.

Reply:
Doesn't the call to SetTensor depends on the existence of padding? Are you suggesting to remove the comment or condition check?

Reply:
SetTensor will only set 0 to input tensor in memory, but since padding is implicit, and doesn't take up any space in memory, so SetTensor is not going to do anything to padding.

You can remove the ```if``` conditional check for padding, everything will just go to ```else``` branch you currently have